File: src/main/java/io/felipeandrade/gsw2/material/GSWItem.java
package io.felipeandrade.gsw2.material;

import net.minecraft.item.Item;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

import io.felipeandrade.gsw2.common.Localizeable;
import io.felipeandrade.gsw2.common.Registrable;

import static io.felipeandrade.gsw2.GSW2Mod.MOD_ID;

public class GSWItem extends Item implements Registrable, Localizeable {

    protected String unlocalizedName;


    public GSWItem(String unlocalizedName, Settings settings) {
        super(settings);
        this.unlocalizedName = unlocalizedName;
    }

    @Override
    public void register() {
        Registry.register(Registry.ITEM, new Identifier(MOD_ID, unlocalizedName), this);
    }

    @Override
    public String unlocalizedName() {
        return unlocalizedName;
    }
}

File: gsw-mod/src/main/kotlin/io/felipeandrade/gsw/item/GSWItem.kt
package io.felipeandrade.gsw.item

import io.felipeandrade.gsw.GSWMod
import io.felipeandrade.gsw.common.Registrable
import net.minecraft.item.Item
import net.minecraft.registry.Registries
import net.minecraft.registry.Registry
import net.minecraft.util.Identifier

open class GSWItem(var unlocalizedName: String, settings: Settings) : Item(settings), Registrable {
    override fun register() {
        Registry.register(Registries.ITEM, Identifier(GSWMod.MOD_ID, unlocalizedName), this)
    }
}